src/4.3_to_5.0/migrateKpi.ts

```diff
@@ -189,7 +190,13 @@ export function migrateKpi(
   const mapping = deriveMappingFromMdx({
     mdx: query.mdx,
     cube,
-    widgetPlugin: pluginWidgetKpi,
+    widgetPlugin: {
+      ...pluginWidgetKpi,
+      // This makes the "allMeasures" tile part of the generated mapping.
+      // This tile indicates the position of the measures on the queries axes, necessary to migrate KPI titles.
+      // It is then removed from the mapping, as Atoti UI 5.0 does not expect it.
```

Comment:
I'm not sure I understand: can't we just recognize for each post-widget-state-migration mapping attribute whether there are measures mapped to it ? And then migrate the titles based on that ?

Reply:
The problem is the position of `ALL_MEASURES` relative to level fields. Indeed, the order of the measure or level identifiers in the tuple matters for KPI titles to work in Atoti UI 5. They have to correspond to their order on the cellset axes.

 But since `pluginWidgetKpi` has `doesSupportMeasuresRedirection: false`, `deriveMappingFromMdx` does not include the `ALL_MEASURES` tile. This prevents us from knowing where the measure unique name should be placed in the tuple.

The simplest solution I found to that problem is to derive the mapping with `doesSupportMeasuresRedirection: true` in order to receive the `ALL_MEASURES` tile and then to filter it out from the mapping before returning the migrated widget state.

Reply:
Thanks for the explanation, it's simple so that sounds good to me if you want to keep it like this.

Just an alternative, we could avoid this trick while keeping things simple IMO:
```
const measuresAxisName = getMeasuresAxisName(mdx);
const positionOfAllMeasuresWithinAxis = getMeasuresPositionOnAxis(
   mdx.axes.find((axis) => axis.name === measuresAxisName)!,
);
```

This is what we do in [`deriveGenericMappingFromMdx`](https://github.com/activeviam/atoti-ui/blob/95e3ec5c9b95775eb5b6a56f465facd38b937d11/packages/widget/src/deriveGenericMappingFromMdx.ts#L152-L155).